internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -559,7 +559,7 @@ func resourceLinuxVirtualMachineCreate(d *pluginsdk.ResourceData, meta interface
 		}
 	}
 
-	if v, ok := d.GetOk("license_type"); ok {
+	if v, ok := d.GetOk("license_type"); ok && v.(string) != "" {
```

Comment:
no need to change this code as `license_type` is an enum type and the value won't be empty
```suggestion
	if v, ok := d.GetOk("license_type"); ok {
```